File: unity/Assets/Scripts/Multiplayer/SuccessResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;
using static Michitai.Multiplayer.Time.Time;

namespace Michitai.Multiplayer
{
    /// <summary>
    /// Standard success response for API operations that don't return specific data in Unity.
    /// Includes a message and timestamp for confirmation with Unity-specific time parsing.
    /// </summary>
    [System.Serializable]
    public class SuccessResponse : ApiResponse<ECommonError>
    {
        [SerializeField]
        private string updated_at;

        /// <summary>
        /// Success message describing the operation result.
        /// </summary>
        public string message;

        /// <summary>
        /// Parsed timestamp when the operation was completed.
        /// Uses Unity-specific UTC parsing.
        /// </summary>
        public DateTimeOffset? UpdatedAt
        {
            get
            {
                return ParseUtc(updated_at);
            }
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/SuccessResponse.java
package com.michitai.multiplayer;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Standard success response for API operations that don't return specific data.
 * Includes a message and timestamp for confirmation.
 */
public class SuccessResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    @JsonProperty("updated_at")
    private String updatedAt;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }
}
